core/src/main/java/com/cloud/storage/template/SwiftVolumeDownloader.java

```diff
@@ -0,0 +1,396 @@
+            LOGGER.error(errorString);
+            status = Status.UNRECOVERABLE_ERROR;
+            return 0;
+        } else {
+            // The ContentLengthHeader is supplied, parse it's value.
+            remoteSize = contentLength;
+        }
+
+        if (remoteSize > maxVolumeSizeInBytes) {
+            errorString = "Remote size is too large for volume " + downloadUrl + " remote size is " + remoteSize + " max allowed is " + maxVolumeSizeInBytes;
+            LOGGER.error(errorString);
+            status = Status.UNRECOVERABLE_ERROR;
+            return 0;
+        }
+
+        InputStream inputStream;
+        try {
+            inputStream = new BufferedInputStream(response.getEntity().getContent());
+        } catch (IOException e) {
+            errorString = "Exception occurred while opening InputStream for volume from " + downloadUrl;
+            LOGGER.error(errorString);
+            status = Status.UNRECOVERABLE_ERROR;
+            return 0;
+        }
+
+        String filePath = installPathPrefix + File.separator + installPath;
+        File directory = new File(filePath);
+        File srcFile = new File(filePath + File.separator + fileName);
+        try {
+            if (!directory.exists()) {
+                LOGGER.info("Creating directories "+filePath);
+                directory.mkdirs();
+            }
+            if (!srcFile.createNewFile()) {
+                LOGGER.info("Reusing existing file " + srcFile.getPath());
+            }
+        } catch (IOException e) {
+            errorString = "Exception occurred while creating temp file " + srcFile.getPath();
+            LOGGER.error(errorString);
+            status = Status.UNRECOVERABLE_ERROR;
+            return 0;
+        }
+
+        LOGGER.info("Starting download from " + downloadUrl + " to staging with size " + remoteSize + " bytes to "+filePath);
+        final Date downloadStart = new Date();
+
+        try (FileOutputStream fileOutputStream = new FileOutputStream(srcFile);) {
+            BufferedOutputStream outputStream = new BufferedOutputStream(fileOutputStream,DOWNLOAD_BUFFER_SIZE_BYTES);
+            byte[] data = new byte[DOWNLOAD_BUFFER_SIZE_BYTES];
+            int bufferLength = 0;
+            while((bufferLength = inputStream.read(data,0,DOWNLOAD_BUFFER_SIZE_BYTES)) >= 0){
+                totalBytes += bufferLength;
+                outputStream.write(data,0,bufferLength);
+                status = Status.IN_PROGRESS;
+                LOGGER.trace("Download in progress: " + getDownloadPercent() + "%");
+                if(totalBytes >= remoteSize){
+                    volumeFile = srcFile;
+                    status = Status.DOWNLOAD_FINISHED;
+                }
+            }
+            outputStream.close();
+            inputStream.close();
+        } catch (IOException e) {
+            LOGGER.error("Exception when downloading from url to staging nfs:" + e.getMessage(), e);
```

Comment:
Suggestion:

Improve logging by adding `downloadUrl`, `remoteSize` and `filePath` to the message.

Reply:
thanks will do it.